src/poseidon.rs

```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+#![cfg(feature = "poseidon")]
+
+#[cfg(feature = "zk")]
+mod zk;
+
+mod zero_items;
+use zero_items::zero_items;
+
+use dusk_bls12_381::BlsScalar;
+use dusk_poseidon::sponge::hash as poseidon_hash;
+
+use crate::Aggregate;
+
+/// The Poseidon Merkle Tree
+pub type Tree<T, const H: usize, const A: usize> = crate::Tree<Item<T>, H, A>;
+
+/// The Poseidon Merkle Opening
+pub type Opening<T, const H: usize, const A: usize> =
+    crate::Opening<Item<T>, H, A>;
+
+/// The Poseidon Node type used for the poseidon merkle tree
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub struct Item<T> {
+    pub hash: BlsScalar,
+    pub data: T,
+}
+
+impl<T> Item<T> {
+    pub fn new(hash: BlsScalar, data: T) -> Self {
```

Comment:
This function should have a doc for example.

Reply:
added small doc